### PR TITLE
UI-2479: Dynamically change the content of the Custom pop up

### PR DESCRIPTION
**Pull request by mallik-svmx**
- Custom/Non-custom and intersecting popups open on single click
- Intersecting popup content to be configurable from the prop, I presume the required route calc handled in application logic.

**Default:**
screen shot 2018-11-01 at 4 18 55 pm

**Dynamic Content Clicking on Intersecting Area:**
screen shot 2018-11-01 at 7 49 08 pm

**Clicking on Non-Intersecting Area:**
screen shot 2018-11-01 at 4 23 55 pm

**Clicking on Intersecting Area:**
screen shot 2018-11-01 at 4 19 24 pm

**Clicking on Isolated route(green):**
screen shot 2018-11-01 at 4 19 34 pm

### CODE REVIEW

**Comment by shao** [comment]
@leslieli @inothings

**Comment by leslieli** [comment]
@mallik-svmx For screenshot 3, you clicked on the Non-Intersecting Area of blue route (nearby the marker 3), right? Did you try to click on Non-Intersecting Area of purple route (nearby marker 1)?

I tried it in my local, the problem is still there. Please see my [comment](https://servicemax.atlassian.net/browse/UI-2479?focusedCommentId=160316&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-160316) in ticket and help check. Thanks.

File: px-map-behavior-layer-geojson.es6.js [hunk 1]
```diff
@@ -181,8 +192,37 @@
...
+          const routeEnd = coords[coords.length - 1];
+          // returns distance (in meters) so to miles * 0.000621371
+          const routeMiles = routeStart.distanceTo(routeEnd) * 0.000621371;
+
+          // TODO: Next Jira Ticket: Clicking on each of these intersecting route will bring up intersectingCustomPopup content
+          return `<span class='intersecting-route-popup'><a href="#">${route.id}: <strong>${routeMiles.toFixed(2)} Miles</a></strong></span>`;
+        });
+        const intersectingRoutePopup = {
+          // fallback to component generated content
+          "content": `<span class='intersecting-route-popup'>${this.intersectingRoutePopup ? this.intersectingRoutePopup : routeContent}</span>`,
```
Comment (leslieli) [inline px-map-behavior-layer-geojson.es6.js]: We are not able to know how many routes are overlap, so we are not able to provide the intersecting route popup content when render the routes. But we can provide and pass each route's info to customPopup property of each feature in geojson. So it is better to get the routes' info from that, then merge the overlap routes' info to one popup when clicking on intersecting area of routes. 

See comment in [jira ticket](https://servicemax.atlassian.net/browse/UI-2479) for detail.

**Comment by mallik-svmx** [comment]
Closing as it's not required anymore.

**Pull request closed without merging**